flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/AsyncCheckpointRunnable.java

```diff
@@ -287,10 +287,9 @@ private void handleExecutionException(Exception e) {
                 } else {
                     // We never decline checkpoint after task is not running to avoid unexpected job
                     // failover, which caused by exceeding checkpoint tolerable failure threshold.
-                    LOG.warn(
-                            "As task is already not running, no longer decline checkpoint {}.",
-                            checkpointMetaData.getCheckpointId(),
-                            checkpointException);
+                    LOG.info(
+                            "Ignore decline of checkpoint {} as task is not running anymore.",
+                            checkpointMetaData.getCheckpointId());
```

Comment:
Can not you keep the `checkpointException`? Or would it also cause e2e test failures?

Reply:
If we still leave `checkpointException` in messages, some e2e tests would fail immediately during exception check, which is not what we expected. Thus, we should remove the checkpoint exception message.